### DISCUSSION TITLE
WebPack 5 Module Federation errors when a module is exposed and running webpack-dev-server

### DISCUSSION THREAD

**Discussion by drewkill32** [webpack/webpack#14064] [Q&A] [answered]
# Bug report

**What is the current behavior?**
WebPack throws errors when a module is exposed through module federation. When I expose a module with the module federation plugin I get the following errors:

>  Uncaught TypeError: __webpack_require__.nmd is not a function
>     at ./node_modules/url/node_modules/punycode/punycode.js (index.js:6)
>     at Object.options.factory (react refresh:6)
>     at __webpack_require__ (bootstrap:24)
>     at fn (hot module replacement:61)
>     at fn (hot module replacement:61)
>     at ./node_modules/url/url.js (url.js:24)
>     at Object.options.factory (react refresh:6)
>     at __webpack_require__ (bootstrap:24)
>     at fn (hot module replacement:61)
>     at fn (hot module replacement:61)

> jsonp chunk loading:27 Uncaught (in promise) ChunkLoadError: Loading chunk src_App_jsx failed.
> (missing: http://localhost:3001/js/src_App_jsx.js)
>     at Object.__webpack_require__.f.j (jsonp chunk loading:27)
>     at ensure chunk:6
>     at Array.reduce (<anonymous>)
>     at Function.__webpack_require__.e (ensure chunk:5)
>     at Function.fn.e (hot module replacement:81)
>     at Function.fn.e (hot module replacement:81)
>     at ./src/index.js (index.js:1)
>     at Object.options.factory (react refresh:6)
>     at __webpack_require__ (bootstrap:24)
>     at __webpack_exec__ (log$:23)

I am exposing a simple module

```js
        exposes: {
          // uncommenting out this line causes an error
          "./index": "./src/index"
        },
```

if I don't expose anything the dev server runs fine.

**If the current behavior is a bug, please provide the steps to reproduce.**

I created a CodeSandbox where I was able to recreate the issue I am having [here](https://codesandbox.io/s/competent-varahamihira-k7xg8?file=/webpack.config.js) line 157

if you comment out 

```
          "./index": "./src/index"
```
the page loads fine. With the line in there, it creates those errors.

**What is the expected behavior?**

no errors

**Other relevant information:**
webpack version:  5.48.0
Node.js version: 14.17.4
Operating System: Win 10
Additional tools:

**Comment by alexander-akait** [comment, marked as answer]
Please provide steps to reproduce

**Reply by alexander-akait** [reply]
Also you have:
- twice `isDevServer && new webpack.HotModuleReplacementPlugin(),` and `devServer: { hot: true, }` (webpack-dev-server automatically inject `HotModuleReplacementPlugin` hwn hot is enabled)

**Reply by alexander-akait** [reply]
Also please update webpack-dev-server to v4

**Reply by alexander-akait** [reply]
Also you don't need `CleanWebpackPlugin`, because we have the `output.clean` option

**Reply by alexander-akait** [reply]
`url-loader` is depreacted and you don't need it anymore in favor assets modules

**Reply by drewkill32** [reply]
Thanks for your help. following those instructions fixed my issue
